### PR TITLE
support ssb-db2

### PR DESCRIPTION
**Pull request by staltz**
@arj03 This module also has no tests, so I'd need a careful review. :pray:

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -64,35 +64,63 @@ class blobsPurge {
...
-      drainGently(
-        {ceiling: this.cpuMax, wait: 60, maxPause: this.maxPause},
-        (msg: Msg) => {
-          if (msg.value.author === this.ssb.id) {
+
+    if (this.ssb.db?.query) {
+      const {and, fullMentions, author, toPullStream} = this.ssb.db.operators;
+      pull(
+        this.ssb.db.query(
+          and(fullMentions!(blobId), author(this.ssb.id)),
```
Comment (arj03) [inline src/index.ts]: I think this would actually be faster without the author here

Comment (arj03) [reply src/index.ts]: also the ! is wierd?

Comment (arj03) [reply src/index.ts]: Ahh now I see the reason for including author

Comment (staltz) [reply src/index.ts]: Alright

**Comment by staltz** [comment]
@arj03 Updated PR

**Comment by arj03** [comment]
There is still the fullMentions!, is that because of typescript?

**Comment by staltz** [comment]
`fullMentions!` with `!` at the end means "ignore TypeScript errors". Basically, TypeScript by default will think that `fullMentions` is undefined, and we're "casting it" to defined here, telling the compiler that "trust me, at this point, I know it's defined".

**Comment by staltz** [comment]
Similar to that, there is `?.` which I really like. It should be landing in a future version of JS: https://github.com/tc39/proposal-optional-chaining

**Review by arj03 — APPROVED** [review]

**Comment by staltz** [comment]
Example: `setTimeout(later, 1000)?.unref()` would work in browsers

Update: correction: `setTimeout(later, 1000).unref?.()`

**Comment by arj03** [comment]
The diff is quite easy to read now. Looks good, funny the require was missing in the README before :)

**Comment by staltz** [comment]
Thanks for the review

**Pull request merged**